src/djlint/formatter/utils/tag.py

```diff
@@ -157,47 +280,123 @@ def __get_tag_style(self, style: str) -> Dict:
+        """
+        Describe if Tag have preformatted text behavior.
+
+        Element preformatted text status is determined from default element styles
+        according to W3C CSS specifications.
+
+        Whatever status is, if ``Tag._is_html`` is ``False`` this will always return
+        ``False``.
+
+        Returns:
+            bool: True if element have a preformatted text behavior else ``False``.
+        """
+        if self._is_html:
+            return self.__css_whitespace.get(
+                tag, self.__css_default_whitespace
+            ).startswith("pre")
+        return False
+
     @property
     def is_space_sensitive(self) -> bool:
-        """Check if tag is space sensitive."""
+        """
+        Check if tag is space sensitive.
+
+        Space sensitivity is based on element CSS flow. If element is HTML, have a
+        block alike flow and not a script element, it is not considered as sensitive.
+
+        Returns:
+            bool: A boolean whether the element is space sensitive or not.
+        """
         if self._is_html:
             display = self.__css_display.get(self.name, self.__css_default_display)
-            # print(display)
+
             return (
                 self.is_script
                 or not display.startswith("table")
                 and display not in ["block", "list-item", "inline-block"]
             )
-        return False
 
-    @property
-    def is_hidden(self) -> bool:
-        if self._is_html:
-            return (
-                self.__css_display.get(self.name, self.__css_default_display) == "none"
-            )
         return False
 
     @property
     def is_indentation_sensitive(self) -> bool:
+        """
+        Check if tag is indentation sensitive.
+
+        Indentation sensitivity is based on element CSS flow. If element is HTML and
+        have a 'pre' alike flow it is considered as sensitive.
+
+        TODO: It seems '__tag_is_pre' have already been used in 'self.is_pre' from
+        '__init__' so it should be safe to rely on this instance attribute instead of
+        running '__tag_is_pre' again each time 'is_indentation_sensitive' is called.
+
+        Returns:
+            bool: A boolean whether the element is indentation sensitive or not.
+        """
         if self._is_html:
             return self.__tag_is_pre(self.name)
+
         return False
 
-    def __tag_is_pre(self, tag: str) -> bool:
+    @property
+    def is_hidden(self) -> bool:
+        """
+        Describe if Tag is an hidden element.
+
+        Element hidden status is determined from element CSS flow (block,
+        inline, table, etc..) as defined in HTML specifications, element CSS style rules
+        is not involved here.
+
+        Whatever hidden status is, if ``Tag._is_html`` is ``False`` this will always
+        return ``False``.
+
+        TODO: This property is a little disturbing with the ``Tag.hidden`` attribute
```

Comment:
we can prob rename this to "is_display_none" or something like that, to better represent what it is looking up in the style.